File: code_blocks/integrations/third-party-integrations/customerio_8.kt
import android.content.Context
import com.revenuecat.purchases.Purchases
import com.revenuecat.purchases.PurchasesConfiguration
import io.customer.sdk.CustomerIO

fun setupRevenueCatWithCustomerIo(context: Context) {
    // Configure RevenueCat first
    val config = PurchasesConfiguration.Builder(context, "public_sdk_key")
        .appUserID("your_app_user_id")
        .build()
    Purchases.configure(config)

    // Ensure Customer.io is initialized and identify has been called
    val customerIoId = CustomerIO.instance().identifier

    customerIoId?.let {
        Purchases.sharedInstance.setAttributes(mapOf("\$customerioId" to it))
    }
}

File: code_blocks/integrations/third-party-integrations/customerio_8.swift
import RevenueCat
import CustomerIO

func setupRevenueCatWithCustomerIo() {
    // Configure RevenueCat first
    Purchases.configure(withAPIKey: "public_sdk_key", appUserID: "your_app_user_id")

    // Ensure Customer.io is initialized and identify has been called
    let customerIoId = CustomerIO.shared.identifier

    if let customerIoId = customerIoId {
        Purchases.shared.attribution.setAttributes([
            "$customerioId": customerIoId
        ])
    }
}

